sdk/core/core-auth/src/tokenCredential.ts

```diff
@@ -49,5 +49,7 @@ export interface AccessToken {
  * @param credential The assumed TokenCredential to be tested.
  */
 export function isTokenCredential(credential: any): credential is TokenCredential {
-  return credential && typeof credential.getToken === "function";
+  return credential
+    && typeof credential.getToken === "function"
+    && credential.signRequest === undefined;
```

Comment:
Perhaps add a comment here as to why the explicit check on `signRequest`? Will help others who have no idea of ms-rest-nodeauth

Reply:
Good point, thanks!

Reply:
Comment added:

```
  // Check for an object with a 'getToken' function but without
  // a 'signRequest' function.  We do this check to make sure that
  // a ServiceClientCredentials implementor (like TokenClientCredentials
  // in ms-rest-nodeauth) doesn't get mistaken for a TokenCredential.
```